core/src/main/java/org/lflang/federated/serialization/FedCustomPythonSerialization.java

```diff
@@ -0,0 +1,165 @@
+/*************
+ * Copyright (c) 2024, The University of California at Berkeley.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+
+package org.lflang.federated.serialization;
+
+import org.lflang.generator.GeneratorBase;
+import org.lflang.target.Target;
+
+/**
+ * Enables support for custom serialization.
+ *
+ * @author Shulu Li
+ */
+public class FedCustomPythonSerialization implements FedSerialization {
+
+  String customSerializerPackage;
+
+  public FedCustomPythonSerialization(String customSerializerPackage) {
+    this.customSerializerPackage = customSerializerPackage;
+  }
+
+  @Override
+  public boolean isCompatible(GeneratorBase generator) {
+    if (generator.getTarget() != Target.Python) {
+      throw new UnsupportedOperationException("This class only support Python serialization.");
```

Comment:
```suggestion
      throw new UnsupportedOperationException("The FedCustomPythonSerialization class only supports the Python target.");
```

Reply:
I don't see my suggested change in the code.  The reason for the suggestion:
* "This class" is a meaningless message to a user.  What class?
* Grammatical error.

Reply:
I agree, this error message is an oversight on my side. I have made your suggested change [here](https://github.com/lf-lang/lingua-franca/pull/2375/files).